### PR TITLE
add support for yecc and leex

### PR DESCRIPTION
**Pull request by lixen-wg2**
Compiles .yrl and .xrl to .erl

### CODE REVIEW

File: private/erlang_xrl_yrl.bzl [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    out_dir = out_dirs[0]
+
+    script = """set -euo pipefail
+
+TMP=$(mktemp -d || mktemp -d -t bazel-tmp)
+test -d "$TMP"
+trap "rm -fr '$TMP'" EXIT
+
+"{erlang_home}"/bin/erlc \\
+    -o {out_dir} \\
```
Comment (jacob-wg2) [inline private/erlang_xrl_yrl.bzl]: Hmm. We could (single) quote this (and the other occurrences, see comment ~above~ below), but we don't do it in `erlang_dia` either (there we have space separated multiple dirs/files, which aren't easy to quote).

File: private/erlang_xrl_yrl.bzl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    out_dirs = unique_dirnames(outputs)
+    if len(out_dirs) > 1:
+        fail(ctx.attr.outs, "do not share a common parent directory")
+    out_dir = out_dirs[0]
+
+    script = """set -euo pipefail
+
+TMP=$(mktemp -d || mktemp -d -t bazel-tmp)
+test -d "$TMP"
+trap "rm -fr '$TMP'" EXIT
```
Comment (jacob-wg2) [inline private/erlang_xrl_yrl.bzl]: We don't need this when using `out_dir` directly.

Should we do a `mkdir -p {out_dir}` for consistency?

File: private/erlang_xrl_yrl.bzl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    out_dir = out_dirs[0]
+
+    script = """set -euo pipefail
+
+
+mkdir -p {out_dir}
+
+"{erlang_home}"/bin/erlc \\
+    -o "{out_dir}" \\
+    $@
```
Comment (jacob-wg2) [inline private/erlang_xrl_yrl.bzl:43]: `"$@"` ?

**Review by jacob-wg2 — APPROVED** [review]
LGTM beside the quoting

**Comment by lixen-wg2** [comment]
Integrated to upstream branch already https://github.com/lixen-wg2/rules_erlang/blob/upstream/private/erlang_xrl_yrl.bzl

**Pull request closed without merging**